File: apps/api/app/schemas/item.py
from __future__ import annotations

from datetime import datetime
from typing import Literal

from pydantic import BaseModel, ConfigDict, Field


class ItemCatalogEntry(BaseModel):
    category: str
    unit: str
    safety_stock_days: int


class HouseholdItemBase(BaseModel):
    name: str | None = None
    category: str
    brand: str | None = None
    spec_text: str | None = None
    unit: str | None = None
    safety_stock_days: int | None = Field(default=None, ge=1, le=30)
    preferred_platform: str | None = None
    replaceable: bool = False
    enabled: bool = True
    is_custom: bool = False


class HouseholdItemCreate(HouseholdItemBase):
    pass


class HouseholdItemUpdate(HouseholdItemBase):
    pass


class HouseholdItemResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    household_id: str
    name: str
    category: str
    brand: str | None
    spec_text: str | None
    unit: str
    safety_stock_days: int
    preferred_platform: str | None
    replaceable: bool
    enabled: bool
    is_custom: bool
    estimated_remaining_days: int | None = None
    inventory_status: Literal["estimated", "estimating", "low_stock", "no_records"] = "no_records"
    last_purchased_at: datetime | None = None

File: packages/shared-types/src/item.ts
import type { InventoryStatus } from "./inventory";

export const SUPPORTED_ITEM_CATEGORIES = [
  "卷纸",
  "抽纸",
  "垃圾袋",
  "洗衣液",
  "洗洁精",
  "牙膏",
  "猫粮",
  "狗粮",
  "猫砂",
  "尿不湿",
  "湿巾",
] as const;

export type SupportedItemCategory = (typeof SUPPORTED_ITEM_CATEGORIES)[number];

export type HouseholdItemFormValues = {
  name: string;
  category: SupportedItemCategory | string;
  brand: string;
  specText: string;
  unit: string;
  safetyStockDays?: number;
  preferredPlatform?: string;
  replaceable?: boolean;
  enabled?: boolean;
  isCustom?: boolean;
};

export type HouseholdItemSummary = {
  id: string;
  name: string;
  category: SupportedItemCategory | string;
  brand: string;
  specText: string;
  enabled: boolean;
  isCustom: boolean;
  estimatedRemainingDays: number | null;
  inventoryStatus: InventoryStatus;
  lastPurchasedAt: string | null;
};

export type HouseholdItemDetail = HouseholdItemSummary & {
  householdId: string;
  unit: string;
  safetyStockDays: number;
  preferredPlatform: string | null;
  replaceable: boolean;
};
